File: easy/Number of 1 Bits/Go/main.go
// Problem: Number of 1 Bits
// Link to the problem: https://leetcode.com/problems/number-of-1-bits/
func hammingWeight(n int) int {
	var ans int = 0
	for n != 0 {
		n &= (n - 1)
		ans++
	}
	return ans
}

File: easy/Number of 1 Bits/Rust/main.rs
// Problem: Number of 1 Bits
// Link to the problem: https://leetcode.com/problems/number-of-1-bits/
impl Solution {
    pub fn hamming_weight(n: i32) -> i32 {
        let mut num: i32 = n;
        let mut ans: i32 = 0;
        while num != 0 {
            num &= (num - 1);
            ans += 1;
        }
        return ans;
    }
}
